### PR TITLE
[LM-9620] Added LMIcon

### PR DESCRIPTION
**Pull request by prabha-likeminds**
## Description

**Features**
- Added LMIcon

**Bugs**

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Hotfix (fixes a breaking changes)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Other, please describe:

### Author Checklist

- [x] Assignee added
- [x] Reviewer added
- [x] Description added
- [ ] Unit test cases added

### Reviewer Checklist

- [x] PR title starts with Jira ticket number prefix
- [ ] If hotfix, Add Hotfix as prefix in PR title
- [ ] Label added
- [x] Commit messages follows standard format (added, changed, removed, code commented and chore)
- [x] Description about the Feature/ Bugs added

### CODE REVIEW

File: src/base/LMIcon/types.tsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+export interface LMIconProps {
+    type: 'png',
+    assetPath?: string,
+    color?: string,
+    size?: number,
+    iconStyle?: ImageStyle,
+    boxFit?: 'center' | 'contain' | 'cover' | 'repeat' | 'stretch',
+    boxStyle?: ViewStyle
+}
```
Comment (ishaan1607) [inline src/base/LMIcon/types.tsx]: add comments for each props, what is represents

File: src/base/LMIcon/types.tsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { ImageStyle, ViewStyle } from "react-native";
+
+export interface LMIconProps {
+    type: 'png',
+    assetPath?: string,
+    color?: string,
+    size?: number,
```
Comment (ishaan1607) [inline src/base/LMIcon/types.tsx]: height and width can be different for a image

File: src/base/LMIcon/index.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+                height: size ? size : defaultStyles.iconStyle.height,
+                tintColor: color,
+                resizeMode: boxFit
+                  ? boxFit
+                  : defaultStyles.iconStyle.resizeMode,
+              },
+            ])}
+          />
+        </View>
+      ) : null}
```
Comment (ishaan1607) [inline src/base/LMIcon/index.tsx]: why `null`

**Review by ishaan1607 — CHANGES_REQUESTED** [review]

File: src/base/LMIcon/index.tsx (hunk 3 above)
Comment (prabha-likeminds) [reply src/base/LMIcon/index.tsx]: resolved

File: src/base/LMIcon/types.tsx (hunk 2 above)
Comment (prabha-likeminds) [reply src/base/LMIcon/types.tsx]: resolved

File: src/base/LMIcon/types.tsx (hunk 1 above)
Comment (prabha-likeminds) [reply src/base/LMIcon/types.tsx]: resolved

File: src/base/LMIcon/types.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+export interface LMIconProps {
+    type: 'png', // this represents the type of icon 
+    assetPath?: string, // this represents the url the icon
+    color?: string, // this represents the tintcolor of the icon
+    height?: number, // this represents the height of the icon
+    width?: number, // this represents the width of the icon
+    iconStyle?: ImageStyle, // this represents the style of the icon
+    boxFit?: 'center' | 'contain' | 'cover' | 'repeat' | 'stretch', // this defines the fit behaviour of the icon inside the box around it
+    boxStyle?: ViewStyle // this represents the style of the view
```
Comment (divyanshgandhilm) [inline src/base/LMIcon/types.ts:11]: Does this cover `backgroundColor`, `borderRadius`, `boxPadding` and `borderSize` all of them?

Comment (prabha-likeminds) [reply src/base/LMIcon/types.ts:11]: Yes

**Review by ishaan1607 — APPROVED** [review]

**Pull request merged**